k8_cortx_cloud/deploy-cortx-cloud.sh

```diff
@@ -841,36 +841,36 @@ function deployCortxSecrets()
     printf "########################################################\n"
     # Parse secret from the solution file and create all secret yaml files
     # in the "auto-gen-secret" folder
-    secret_auto_gen_path="$cfgmap_path/auto-gen-secret-$namespace"
-    mkdir -p $secret_auto_gen_path
+    secret_auto_gen_path="${cfgmap_path}/auto-gen-secret-${namespace}"
+    mkdir -p ${secret_auto_gen_path}
     secret_name=$(parseSolution "solution.secrets.name")
-    secret_fname=$(echo $secret_name | cut -f2 -d'>')
-    yaml_content_path=$(echo $secret_name | cut -f1 -d'>')
+    secret_fname=$(echo ${secret_name} | cut -f2 -d'>')
+    yaml_content_path=$(echo ${secret_name} | cut -f1 -d'>')
     yaml_content_path=${yaml_content_path/.name/".content"}
-    secrets="$(./parse_scripts/yaml_extract_block.sh $solution_yaml $yaml_content_path 2)"
+    secrets="$(./parse_scripts/yaml_extract_block.sh ${solution_yaml} ${yaml_content_path} 2)"
 
-    new_secret_gen_file="$secret_auto_gen_path/$secret_fname.yaml"
-    cp "$cfgmap_path/templates/secret-template.yaml" $new_secret_gen_file
-    ./parse_scripts/subst.sh $new_secret_gen_file "secret.name" "$secret_fname"
-    ./parse_scripts/subst.sh $new_secret_gen_file "secret.content" "$secrets"
+    new_secret_gen_file="${secret_auto_gen_path}/${secret_fname.yaml}"
+    cp "${cfgmap_path}/templates/secret-template.yaml" ${new_secret_gen_file}
+    ./parse_scripts/subst.sh ${new_secret_gen_file} "secret.name" "${secret_fname}"
+    ./parse_scripts/subst.sh ${new_secret_gen_file} "secret.content" "${secrets}"
         
-    kubectl_cmd_output=$(kubectl create -f $new_secret_gen_file --namespace=$namespace 2>&1)
+    kubectl_cmd_output=$(kubectl create -f ${new_secret_gen_file} --namespace=${namespace} 2>&1)
 
-    if [[ "$kubectl_cmd_output" == *"BadRequest"* ]]; then
-        printf "Exit early. Create secret failed with error:\n$kubectl_cmd_output\n"
+    if [[ "${kubectl_cmd_output}" == *"BadRequest"* ]]; then
+        printf "Exit early. Create secret failed with error:\n%s\n" "${kubectl_cmd_output}"
         exit 1
     fi
-    echo $kubectl_cmd_output
+    echo ${kubectl_cmd_output}
 
     control_secret_path="./cortx-cloud-helm-pkg/cortx-control/secret-info.txt"
     data_secret_path="./cortx-cloud-helm-pkg/cortx-data/secret-info.txt"
     server_secret_path="./cortx-cloud-helm-pkg/cortx-server/secret-info.txt"
     ha_secret_path="./cortx-cloud-helm-pkg/cortx-ha/secret-info.txt"
 
-    printf "$secret_fname" >> $control_secret_path
-    printf "$secret_fname" >> $data_secret_path
-    printf "$secret_fname" >> $server_secret_path
-    printf "$secret_fname" >> $ha_secret_path
+    printf "${secret_fname}" >> ${control_secret_path}
+    printf "${secret_fname}" >> ${data_secret_path}
+    printf "${secret_fname}" >> ${server_secret_path}
+    printf "${secret_fname}" >> ${ha_secret_path}
```

Comment:
Sorry to be picky, but can you fix these `printf` as well?

```suggestion
    printf "%s" "${secret_fname}" >> ${control_secret_path}
    printf "%s" "${secret_fname}" >> ${data_secret_path}
    printf "%s" "${secret_fname}" >> ${server_secret_path}
    printf "%s" "${secret_fname}" >> ${ha_secret_path}
```

Codacy doesn't flag them, but our Github action [does](https://github.com/Seagate/cortx-k8s/runs/5100439548?check_suite_focus=true).

```
./k8_cortx_cloud/deploy-cortx-cloud.sh:870:12: note: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
./k8_cortx_cloud/deploy-cortx-cloud.sh:871:12: note: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
./k8_cortx_cloud/deploy-cortx-cloud.sh:872:12: note: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
./k8_cortx_cloud/deploy-cortx-cloud.sh:873:12: note: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
```

Reply:
Yes, of course!  I'll address this in an hour or so.